build/Targets/Settings.targets

```diff
@@ -133,18 +133,18 @@
        Basically, we check for a registry key that will only exist in legacy VS installs, and assume
        we are a willow based installation if our VSVersion is 15.0 and the registry key doesn't exist. -->
   <PropertyGroup>
-    <VisualStudioInstallDirectoryFromRegistry Condition="'$(OS)' == 'Windows_NT'">$(registry:HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\VisualStudio\15.0@InstallDir)</VisualStudioInstallDirectoryFromRegistry>
-    <VisualStudioInstallDirectoryFromRegistry Condition="'$(OS)' == 'Windows_NT' AND '$(VisualStudioInstallDirectoryFromRegistry)' == ''">$(registry:HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\VisualStudio\15.0@InstallDir)</VisualStudioInstallDirectoryFromRegistry>
+    <IsDev15BeforePreview4 Condition="'$(OS)' == 'Windows_NT'">$(registry:HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\VisualStudio\15.0@InstallDir)</IsDev15BeforePreview4>
+    <IsDev15BeforePreview4 Condition="'$(OS)' == 'Windows_NT' AND '$(IsDev15BeforePreview4)' == ''">$(registry:HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\VisualStudio\15.0@InstallDir)</IsDev15BeforePreview4>
```

Comment:
My suggestion is to simply make this a Boolean-valued property, i.e.:

    <IsDev15BeforePreview4 Condition="$(registry:HKEY...)' != ''>true</IsDev15BeforePreview4>

Then you just have two cases that can set it to true, you don't have to worry about the ordering, etc.